ts/WoltLabSuite/Core/Ui/User/ObjectWatch.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Handles the object watch button.
+ *
+ * @author	Marcel Werk
+ * @copyright	2001-2022 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/Ui/User/ObjectWatch
+ * @since 6.0
+ */
+
+import * as Ajax from "../../Ajax";
+import * as UiNotification from "../Notification";
+import * as Language from "../../Language";
+import * as EventHandler from "../../Event/Handler";
+
+const dropdowns = new Map<number, Set<HTMLElement>>();
+
+async function click(element: HTMLElement): Promise<void> {
+  const dropdown = element.closest(".userObjectWatchDropdown") as HTMLElement;
+  const subscribe = parseInt(element.dataset.subscribe!, 10);
+  const objectID = parseInt(dropdown.dataset.objectId!, 10);
+  const objectType = dropdown.dataset.objectType!;
+
+  await Ajax.dboAction("saveSubscription", "wcf\\data\\user\\object\\watch\\UserObjectWatchAction")
+    .payload({
+      enableNotification: 1,
+      objectID,
+      objectType,
+      subscribe,
+    })
+    .dispatch();
+
+  if (dropdowns.has(objectID)) {
+    dropdowns.get(objectID)!.forEach((element) => {
+      element.querySelectorAll(".userObjectWatchSelect").forEach((li) => {
+        li.classList.remove("active");
+      });
+
+      element.querySelector(`.userObjectWatchSelect[data-subscribe="${subscribe}"]`)?.classList.add("active");
```

Comment:
This can be merged with the `forEach` block above, you are iterating over all elements anyway instead of just the `.active` elements.